OloEngine/src/OloEngine/Asset/AssetImporter.cpp

```
@@ -44,13 +44,29 @@ namespace OloEngine
+    // manager — leaving "No serializer available for asset type" failures. The
+    // registry must live as long as ANY manager references it.
+    static std::atomic<i32> s_InitRefCount{ 0 };
     static std::atomic<bool> s_IsShuttingDown{ false };
 
     void AssetImporter::Init()
     {
-        std::call_once(s_InitFlag, []()
-                       {
+        // A manager is coming up — clear any prior "shutting down" latch so a
+        // registry that was fully torn down can be repopulated below.
+        s_IsShuttingDown.store(false, std::memory_order_release);
```

Cppcheck (2.20.0)

[style] The function 'IsLoading' is never used.

(unusedFunction)